docs/site/Transactions.md

```diff
@@ -0,0 +1,201 @@
+---
+title: 'Using database transactions'
+lang: en
+layout: page
+keywords: LoopBack 4.0, LoopBack 4, Transactions, Transaction
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Using-database-transactions.html
+summary: Transactional API usage in LoopBack 4
+---
+
+## Overview
+
+A *transaction* is a sequence of data operations performed as a single logical
+unit of work. Many relational databases support transactions to help enforce
+data consistency and business logic requirements.
+
+A repository can perform operations in a transaction when the backing datasource
+is attached to one of the following connectors:
+
+- [MySQL connector](MySQL-connector.html) (IMPORTANT: Only with InnoDB as the
+  storage engine).
+- [PostgreSQL connector](PostgreSQL-connector.html)
+- [SQL Server connector](SQL-Server-connector.html)
+- [Oracle connector](Oracle-connector.html)
+- [DB2 Connector](DB2-connector.html)
+- [DashDB Connector](DashDB.html)
+- [DB2 iSeries Connector](DB2-iSeries-connector.html)
+- [DB2 for z/OS connector](DB2-for-z-OS.html)
+- [Informix connector](Informix.html)
+
+The repository class needs to extend from `TransactionalRepository` repository
+interface which exposes the `beginTransaction()` method.
+
+## Transaction APIs
+
+The `@loopback/repository` package includes `TransactionalRepository` interface
+based on `EntityCrudRepository` interface. The `TransactionalRepository`
+interface adds a `beginTransaction()` API that, for connectors that allow it,
+will start a new Transaction. The `beginTransaction()` function gives access to
+the lower-level transaction API, leaving it up to the user to create and manage
+transaction objects, commit them on success or roll them back at the end of all
+intended operations. See [Handling Transactions](#handling-transactions) below
+for more details.
+
+## Handling Transactions
+
+See
+the [API reference](https://apidocs.strongloop.com/loopback-datasource-juggler/#datasource-prototype-begintransaction) for
+full transaction lower-level API documentation.
+
+Performing operations in a transaction typically involves the following steps:
+
+- Start a new transaction.
+- Perform create, read, update, and delete operations in the transaction.
+- Commit or rollback the transaction.
+
+### Start transaction
+
+Use the `beginTransaction()` method to start a new transaction.
+
+Here is an example:
+
+```typescript
+import {
+  Transaction,
+  DefaultTransactionalRepository,
+  IsolationLevel,
+} from '@loopback/repository';
+// assuming there is a Note model extending Entity class, and
+// crudDs datasource which is backed by a transaction enabled
+// connector
+const repo = new DefaultTransactionalRepository(Note, crudDs);
+// Now we have a transaction (tx)
+const tx: Transaction = await repo.beginTransaction({
```

Comment:
Is the explicit type required? If it's your intention to make the type of `tx` explicitly clear to readers, then I am fine with that.

Otherwise the following code should work equally well as far as the compiler is concerned:

```suggestion
const tx = await repo.beginTransaction({
```

Reply:
Yeah it was just to make it explicit. I don't mind removing it too.